app/controllers/api/dishes_controller.rb

```diff
@@ -75,22 +76,29 @@ def user_not_authorized
              status: :unauthorized
     end
 
+    def user_unprocessable_entity
+      render json: {error: {dish: {message: 'Debt too large for some users',
+                                   limit: Dish::MAX_DEBT}}},
+             status: :unprocessable_entity
+    end
+
     def generate_user_dishes(dish)
       dish.user_dishes.create!([
-        {
-          user: current_user,
-          dish_owner: true,
-        },
-        *params[:user_ids].map { |user_id| {user_id: user_id} },
-      ])
+                                 {
+                                   user: current_user,
+                                   dish_owner: true,
+                                 },
+                                 *params[:user_ids]&.map do |user_id|
+                                   {user_id: user_id}
+                                 end,
+                               ])
     end
 
     def validate_users_debts
-      params[:user_ids]&.each do |user_id|
-        if User.find(user_id).total_debt.to_i < Dish::MAX_DEBT
-          user_not_authorized
-        end
-      end
+      return if User.where(id: params[:user_ids])
```

Comment:
nit:
I find this line break confusing. If it's not fitting in 80 chars I would extract `User.where(id: params[:user_ids]).none? { |user| user.total_debt.to_i < Dish::MAX_DEBT }` to a variable.